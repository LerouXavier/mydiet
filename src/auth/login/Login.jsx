import React, {useState} from 'react';
import PropTypes from 'prop-types';

import {isEmpty as _isEmpty} from 'lodash';

import {Form, Message} from 'semantic-ui-react';

export const Login = () => {
    const [credentials, setCredentials] = useState({
        email: '',
        password: ''
    });

    return (
        <Form error={true}>
            <Form.Input
                className="nu-login__email"
                label="E-mail"
                onChange={(event, element) => setCredentials({email: element.value})}
                type="email"
            />
            <Form.Input
                className="nu-login__password"
                label="Password"
                onChange={(event, element) => setCredentials({password: element.value})}
                type="password"
            />
            <Form.Button
                className="nu-login__submit"
                disabled={false}
                floated="right"
                positive={true}
            >
                Login
            </Form.Button>
        </Form>
    );
}