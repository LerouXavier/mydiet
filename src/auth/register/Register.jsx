import * as React from 'react';
import {connect} from 'react-redux';

import {isEmpty as _isEmpty} from 'lodash';

import {Form} from 'semantic-ui-react';

import {authActions} from '../AuthActions';

export class Register extends React.Component {
    state = {
        firstName: '',
        lastName: '',
        language: 'polish',
        email: '',
        password: ''
    };

    render() {
        const {submitForm} = this.props;
        return (
            <React.Fragment>
                <Form.Input
                    className="nu-register__first-name"
                    label="Imię"
                    onChange={(event, element) => this.setState({firstName: element.value})}
                    type="text"
                />
                <Form.Input
                    className="nu-register__last-name"
                    label="Nazwisko"
                    onChange={(event, element) => this.setState({lastName: element.value})}
                    type="text"
                />
                <Form.Input
                    className="nu-register__email"
                    label="Wprowaź adres E-mail"
                    onChange={(event, element) => this.setState({email: element.value})}
                    required={true}
                    type="email"
                />
                <Form.Input
                    className="nu-register__password"
                    label="Hasło"
                    onChange={(event, element) => this.setState({password: element.value})}
                    required={true}
                    type="password"
                />
                <Form.Button
                    className="nu-register__submit"
                    disabled={_isEmpty(this.state.email) || _isEmpty(this.state.password)}
                    floated="right"
                    onClick={() => submitForm(this.state)}
                    positive={true}
                >
                    Zarejestruj
                </Form.Button>
            </React.Fragment>
        );
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        submitForm: (credentials) => {
            dispatch(authActions.register(credentials));
        }
    };
};

export default connect(null, mapDispatchToProps)(Register);