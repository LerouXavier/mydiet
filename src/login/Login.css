#myVideo {
    position: fixed;
    right: 0;
    bottom: 0;
    min-width: 100%;
    min-height: 100%;
}

.np-login {
    height: 100vh;
}
.np-login__form {
    padding: 50px;
}
.np-login__title {
    font-family: 'Playfair Display', serif;
    font-style: italic;
    font-weight: bold;
    font-size: 54px;
    margin-bottom: 37px;
    margin-top: 23px;
}