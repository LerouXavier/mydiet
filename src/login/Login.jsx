import * as React from 'react';
import {Redirect} from 'react-router-dom';
import {isEmpty as _isEmpty} from 'lodash';

import {Form, Header, Grid, Segment} from 'semantic-ui-react';

import {authenticationService} from '../auth/AuthenticationService';
import video from './Bokeh Video Of Leaves.mp4';
import './Login.css';

export class Login extends React.Component {
    state = {
        email: '',
        password: ''
    };

    render() {
        // const {location, user} = this.props;
        // const {from} = location.state || {from: {pathname: '/'}};
        //
        // if (user !== null) {
        //     return <Redirect to={from}/>;
        // }

        return (
            <React.Fragment>
                <video autoPlay muted loop id="myVideo">
                    <source src={video} type="video/mp4"/>
                </video>

                <Grid centered={true} verticalAlign="middle" columns={4} stretched={true} className="np-login">
                    <Grid.Column>
                        <Segment as={Form} basic={true} inverted={true} padded="very" className="np-login__form np-form">
                            <Header as="h1" icon={true} textAlign="center" inverted={true}>
                                <Header.Content className="np-login__title">Nutritional Plan</Header.Content>
                            </Header>
                            <Form.Input
                                label="Wprowaź adres E-mail"
                                type="email"
                                onChange={(event, element) => this.setState({email: element.value})}
                            />
                            <Form.Input
                                label="Hasło"
                                type="password"
                                onChange={(event, element) => this.setState({password: element.value})}
                            />
                            <Form.Button
                                onClick={() => authenticationService.login(this.state)}
                                disabled={_isEmpty(this.state.email) || _isEmpty(this.state.password)}
                                positive={true}
                                floated="right"
                            >
                                Sign In
                            </Form.Button>
                        </Segment>
                    </Grid.Column>
                </Grid>
            </React.Fragment>
        );
    }
}