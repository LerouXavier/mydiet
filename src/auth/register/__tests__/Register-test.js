import renderer from 'react-test-renderer';
import {Register} from '../Register';
import React from 'react';
import {shallow} from 'enzyme/build';
import {Login} from '../../login/Login';

describe('Register', () => {
    test('should render correctly', function () {
        const register = renderer.create(<Register/>).toJSON();
        expect(register).toMatchSnapshot();
    });

    test('should change components state after setting form elements', function () {
        const login = shallow(<Login/>);

        login
            .find('.nu-login__email')
            .simulate('change', {}, {value: 'example@ex.com'});

        expect(login.instance().state.email).toEqual('example@ex.com');
    });
});