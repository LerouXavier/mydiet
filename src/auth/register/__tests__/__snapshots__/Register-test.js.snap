// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
Array [
  <div
    className="field"
>
    <label
        htmlFor={undefined}
    >
        Imię
    </label>
    <div
        className="ui input"
        content={undefined}
    >
        <input
            disabled={undefined}
            onChange={[Function]}
            required={undefined}
            tabIndex={undefined}
            type="text"
        />
    </div>
</div>,
  <div
    className="field"
>
    <label
        htmlFor={undefined}
    >
        Nazwisko
    </label>
    <div
        className="ui input"
        content={undefined}
    >
        <input
            disabled={undefined}
            onChange={[Function]}
            required={undefined}
            tabIndex={undefined}
            type="text"
        />
    </div>
</div>,
  <div
    className="required field"
>
    <label
        htmlFor={undefined}
    >
        Wprowaź adres E-mail
    </label>
    <div
        className="ui input"
        content={undefined}
    >
        <input
            disabled={undefined}
            onChange={[Function]}
            required={true}
            tabIndex={undefined}
            type="email"
        />
    </div>
</div>,
  <div
    className="required field"
>
    <label
        htmlFor={undefined}
    >
        Hasło
    </label>
    <div
        className="ui input"
        content={undefined}
    >
        <input
            disabled={undefined}
            onChange={[Function]}
            required={true}
            tabIndex={undefined}
            type="password"
        />
    </div>
</div>,
  <div
    className="disabled field"
>
    <button
        className="ui positive disabled right floated button"
        disabled={true}
        onClick={[Function]}
        required={undefined}
        role="button"
        tabIndex={-1}
        type={undefined}
    >
        Zarejestruj
    </button>
</div>,
]
`;
